packages/tradershub/src/features/cfd/modals/TopUpModal/TopUpModal.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { useIsEuRegion, useMT5Deposit, useOtherCFDPlatformsDeposit } from '@deriv/api';
+import { Button, Heading, Text, useBreakpoint } from '@deriv/quill-design';
+import { Modal } from '../../../../components/Modal';
+import { THooks, TPlatforms } from '../../../../types';
+import { CFDPlatforms, MarketType, MarketTypeDetails, PlatformDetails } from '../../constants';
+
+type TTopUpModalProps = {
+    account: THooks.CtraderAccountsList | THooks.DxtradeAccountsList | THooks.MT5AccountsList;
+    platform: TPlatforms.All;
+};
+
+const TopUpModal = ({ account, platform }: TTopUpModalProps) => {
+    const { isDesktop } = useBreakpoint();
+    const { data: isEuRegion } = useIsEuRegion();
+    const { mutateAsync: MT5Deposit } = useMT5Deposit();
+    const { mutateAsync: OtherCFDPlatformsDeposit } = useOtherCFDPlatformsDeposit();
+
+    const topUpVirtual = async () => {
+        if (platform === CFDPlatforms.MT5) {
+            await MT5Deposit({
+                to_mt5: account?.login ?? '',
+            });
+        } else {
+            await OtherCFDPlatformsDeposit({
+                platform,
+                to_account: (account as THooks.CtraderAccountsList | THooks.DxtradeAccountsList).account_id ?? '',
+            });
+        }
+    };
+
+    const platformTitle = PlatformDetails[platform].title;
+    const marketTypeDetails = MarketTypeDetails(isEuRegion)[account.market_type ?? MarketType.ALL];
+    const marketTypeTitle = marketTypeDetails?.title ?? '';
+    const title = platform === CFDPlatforms.MT5 ? `${platformTitle} ${marketTypeTitle}` : platformTitle;
+
+    const HeadingTag = isDesktop ? Heading.H3 : Heading.H2;
+    const balance =
+        platform === CFDPlatforms.CTRADER
+            ? (account as THooks.CtraderAccountsList)?.formatted_balance
```

Comment:
@sergei-deriv Why for CTrader we need currency beside balance?

Reply:
@mahdiyeh-deriv I did it because mt5 and dxtrade accounts have `display_balance` created in the hooks with balance and currency, but for cCtrader accounts we have `display_balance` from BE (just balance value) and `formatted_balance` created inside of the hook (balance and currency)

Reply:
@sergei-deriv yeah, we have the same `display_balance` key coming from json, we need to change this in to formatted_balance in `mt5` and `derivx` hooks, but this is out of the scope of this PR.